File: examples/dart/json_parser.dart
// Minimal JSON configuration parser used to demonstrate consuming the
// json2vars-setter matrix file in a Dart project. It uses only the Dart SDK
// (dart:convert), so no package dependencies are required.

import 'dart:convert';
import 'dart:io';

/// Parse [contents] (JSON text) into a map. Returns `null` on parse failure.
Map<String, dynamic>? parseConfig(String contents, {bool silent = false}) {
  try {
    return jsonDecode(contents) as Map<String, dynamic>;
  } catch (e) {
    if (!silent) {
      stderr.writeln('Error parsing JSON: $e');
    }
    return null;
  }
}

void main() {
  final file = File('dart_project_matrix.json');
  final result = parseConfig(file.readAsStringSync());
  if (result != null) {
    print(const JsonEncoder.withIndent('  ').convert(result));
  }
}

File: examples/julia/json_parser.jl
# Minimal JSON configuration parser used to demonstrate consuming the
# json2vars-setter matrix file in a Julia project. Parsing uses the JSON.jl
# package declared in Project.toml.

using JSON

"""
    parse_config(contents; silent=false)

Parse `contents` (JSON text) into a `Dict`. Returns `nothing` on parse failure.
"""
function parse_config(contents::AbstractString; silent::Bool = false)
    try
        return JSON.parse(contents)
    catch err
        silent || println(stderr, "Error parsing JSON: ", err)
        return nothing
    end
end

if abspath(PROGRAM_FILE) == @__FILE__
    result = parse_config(read("julia_project_matrix.json", String))
    result === nothing || println(JSON.json(result, 2))
end
